src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let joinedString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === 0 && (this[i] === null || this[i] === undefined)) {
+        continue;
+      } else if (this[i] === null || this[i] === undefined) {
+        joinedString += `${separator}`;
+      } else if (i === 0) {
+        joinedString = `${this[0]}`;
+      } else {
+        joinedString += `${separator}` + `${this[i]}`;
+      }
```

Comment:
those condition seems to be little complex, please try something like this:
```suggestion
      if ((this[i] !== null && this[i] !== undefined)) {
          // add letter to joinedString
        }
        
      if (iterator is less than word length) // add separator to joinedString
```